src/main/java/org/pac4j/vertx/VertxWebContext.java

```diff
@@ -126,24 +130,40 @@ public String getRequestHeader(String name) {
 
   @Override
   public void setSessionAttribute(String name, Object value) {
-    sessionAttributes.getCustomAttributes().put(name, value);
+    Session session = routingContext.session();
```

Comment:
Yes, and I think we can build custom session providers on the vert.x side if we want (for example if you want to persist to Redis, so this is a concern that as far as possible we should keep out of pac4j implementations).